### ISSUE TITLE
NodeJS dbus error 

### ISSUE THREAD

**Issue by MarcosPer** [JasonLG1979/gnome-shell-extensions-mediaplayer#416]
Hi, I'm making a fork of https://github.com/Quacky2200/Spotify-Web-Player-for-Linux that is an simple  Electron and NodeJS app for spotify web player. And i can't solve this problem:

![captura de pantalla de 2017-08-16 21-50-05](https://user-images.githubusercontent.com/19536941/29382478-f195ed8e-82cc-11e7-8651-a18bce6752cd.png)

It doesn't recive any data but buttons are working.

When song change it sent this metadata.

![captura de pantalla de 2017-08-16 21-44-30](https://user-images.githubusercontent.com/19536941/29382547-25824c32-82cd-11e7-9f48-9d4bb46f343e.png)

The data are sent using dbus library and mpris-service of nodeJS but i can't send any data.

Anyone have an idea of what are wrong? The extension works great with other players in my computer like Ryhtmbox

In summary, I can't send data from the NodeJS

Thanks in advance

**Comment by MarcosPer** [comment]
Using standalone mpris-service code it doesn't work 

Example:  https://github.com/emersion/mpris-service/blob/master/examples/player.js

**Comment by JasonLG1979** [comment]
I have no experience with that module. But if I had to guess it looks like you're trying to send one big object as the metadata. That won't work. Each metadata field must be a variant types packed in an array of string variants.

Looking here:
https://github.com/emersion/mpris-service/blob/master/index.js#L273-L277

It looks like it tries to take care of the last part (packing the values in an array of string variants).


You're better off just writing your own interface so you know it's done right than depending on someone else's code that apparently doesn't work?

**Comment by JasonLG1979** [comment]
For further assistance I suggest opening an issue at https://github.com/emersion/mpris-service
